src/main/java/com/redhat/restdemo/controllers/AuthorController.java

```diff
@@ -1,36 +1,99 @@
 package com.redhat.restdemo.controllers;
 
 import com.redhat.restdemo.model.entity.Author;
+import com.redhat.restdemo.model.entity.Book;
 import com.redhat.restdemo.model.service.AuthorService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
 
 
 @RestController
+@RequestMapping("/author")
 public class AuthorController {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorController.class);
     @Autowired
     private AuthorService authorService;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorController.class);
-
-    @GetMapping("/authors")
-    public ResponseEntity<Iterable<Author>> listAuthors() {
+    @GetMapping
+    public ResponseEntity<Iterable<Author>> getAllAuthors() {
         LOGGER.info("Just hit author endpoint!");
         Iterable<Author> authors = authorService.findAll();
         return new ResponseEntity<>(authors, HttpStatus.OK);
     }
 
-    @PostMapping("/author/add")
+    @GetMapping("/{id}")
+    public ResponseEntity<Optional<Author>> getAuthorById(@PathVariable Integer id) {
+        Optional<Author> author = authorService.findAuthorById(id);
+        if (author.isEmpty()) {
+            LOGGER.info("Author with given ID does not exists");
+            return new ResponseEntity<>(author, HttpStatus.NOT_FOUND);
+        }
+        LOGGER.info("Author found!");
+        return new ResponseEntity<>(author, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}/books")
+    public ResponseEntity<Iterable<Book>> getBooksByAuthor(@PathVariable Integer id) {
+        Iterable<Book> author = authorService.findBooksByAuthor(id);
+        LOGGER.info("Books found!");
+        return new ResponseEntity<>(author, HttpStatus.OK);
+    }
+
+    @PostMapping("/add")
     public ResponseEntity<Author> addAuthor(@RequestBody Author author) {
-        Author createdAuthor =  authorService.addAuthor(author);
+        Author createdAuthor = authorService.addAuthor(author);
+        LOGGER.info("Author successfully added!");
         return new ResponseEntity<>(createdAuthor, HttpStatus.CREATED);
     }
+
+    @PostMapping("/{id}/books")
+    public ResponseEntity<Book> addBookToAuthor(@RequestBody Book book, @PathVariable Integer id) {
+        Book addedBook = authorService. addBookToAuthor(id, book);
+        if (addedBook == null) {
+            LOGGER.info("Please, specify book's ISBN");
```

Comment:
Are you sure that the problem is an empty ISBN? I think that this endpoint would not even work without a fully specified book because you have all attributes by default marked as NotNull.